wagtail/core/models/sites.py

```diff
@@ -140,20 +142,66 @@ def clean_fields(self, exclude=None):
                     ]}
                 )
 
-    @staticmethod
-    def get_site_root_paths():
+    @cached_property
+    def default_root_path(self) -> SiteRootPath:
         """
-        Return a list of `SiteRootPath` instances, most specific path
-        first - used to translate url_paths into actual URLs with hostnames
+        .. versionadded:: 2.16
+
+        Returns the 'default' root path for this site, indicating where
+        URLs 'start' for the selected ``root_page``.
 
-        Each root path is an instance of the `SiteRootPath` named tuple,
-        and have the following attributes:
+        Root paths are used in page routing and URL generation logic, and are
+        provided as ``SiteRootPath`` named tuple instance, which have the
+        following attributes:
 
         - `site_id` - The ID of the Site record
         - `root_path` - The internal URL path of the site's home page (for example '/home/')
         - `root_url` - The scheme/domain name of the site (for example 'https://www.example.com/')
         - `language_code` - The language code of the site (for example 'en')
         """
+        return SiteRootPath(self.id, self.root_page.url_path, self.root_url, self.root_page.locale.language_code)
+
+    def get_root_paths(self) -> List[SiteRootPath]:
+        """
+        .. versionadded:: 2.16
+
+        Returns a list of all known root paths for this site.
+
+        If using Wagtail's built-in internationalisation features, a
+        root path will be included for every known translation of the
+        the site's ``root_page``.
+        """
+        values = [self.default_root_path]
+        if getattr(settings, 'WAGTAIL_I18N_ENABLED', False):
+            for root_page in self.root_page.get_translations().select_related('locale').order_by("path"):
```

Comment:
Note to reviewer: I've moved from using `get_translations(inclusive=True)` to `self.default_root_path` for the non-translated value, as I felt it made things easier to follow. In most cases, `root_page` and `root_page.locale` are prefetched, so this shouldn't result in additional queries.